### PR TITLE
global: admin views for invenio-collections

### PR DESCRIPTION
**Pull request by bruno314**
- Adds admin views for managing invenio-collections.

Signed-off-by: Bruno Cuc bruno.cuc@cern.ch

### CODE REVIEW

**Comment by bruno314** [comment]
I broke my local repo, sorry. Previous PR comments are incorporated, except for `\n\0` EOF (still figuring that one out).

File: invenio_collections/admin.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+
+class CollectionView(ModelView):
+    """Flask-Admin view to manage users."""
+
+    can_view_details = True
+    can_delete = False
+
+    list_all = (
+        'id', 'name', 'right', 'left', 'parent', 'dbquery', 'level', 'tree_id'
```
Comment (jirikuncar) [inline invenio_collections/admin.py]: usually people expect `left` before `right`.

File: invenio_collections/admin.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        column_sortable_list = \
+        column_details_list = \
+        list_all
+
+    column_filters = ('id', 'name', 'tree_id', 'level', 'parent')
+
+    column_default_sort = ('left', True)
+
+    column_labels = {
+        'dbquery': _('Database query'),
```
Comment (jirikuncar) [inline invenio_collections/admin.py]: `Definition` or `Search Query`

File: invenio_collections/admin.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+from flask import current_app, flash
+from flask_admin.contrib.sqla import ModelView
+from flask_admin.form.fields import DateTimeField
+from flask_login import login_user, logout_user
+from werkzeug.local import LocalProxy
+
+from .models import Collection
+
+
+def _(x):
```
Comment (jirikuncar) [inline invenio_collections/admin.py]: `"""Identity."""`

File: invenio_collections/cli.py [hunk 4]
```diff
@@ -49,6 +49,7 @@ def inner(dry_run, *args, **kwargs):
         else:
             db.session.rollback()
         return ret
+
```
Comment (jirikuncar) [inline invenio_collections/cli.py]: why?

File: invenio_collections/models.py [hunk 5]
```diff
@@ -69,5 +69,4 @@ def collection_attribute_changed(target, value, oldvalue, initiator):
 listen(Collection.dbquery, 'set', collection_attribute_changed)
 listen(Collection.parent_id, 'set', collection_attribute_changed)
 
-
-__all__ = ('Collection', )
+__all__ = ('Collection',)
```
Comment (jirikuncar) [inline invenio_collections/models.py]: keep space after `,`

File: invenio_collections/query.py [hunk 6]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 #
 # This file is part of Invenio.
-# Copyright (C) 2015 CERN.
+# Copyright (C) 2015, 2016 CERN.
```
Comment (jirikuncar) [inline invenio_collections/query.py]: no update in file no need to update copyright!

File: tests/test_percolator.py [hunk 7]
```diff
@@ -80,7 +80,7 @@ def _try_populate_collections():
         'properties': {
                 'title': {'type': 'string'},
                 'field': {'type': 'boolean'},
-            },
+        },
```
Comment (jirikuncar) [inline tests/test_percolator.py]: if you are fixing indentation then fix also lines above

Comment (bruno314) [inline tests/test_percolator.py]: fixed

File: invenio_collections/admin.py (hunk 3 above)
Comment (bruno314) [inline invenio_collections/admin.py]: fixed

File: tests/test_admin.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+def test_admin(app):
+    """Test flask-admin interace."""
+    InvenioCollections(app)
+
+    assert isinstance(collection_adminview, dict)
+
+    assert 'model' in collection_adminview
+    assert 'modelview' in collection_adminview
+
+    admin = Admin(app, name="Test")
```
Comment (jirikuncar) [inline tests/test_admin.py]: `'Test'`

**Comment by jirikuncar** [comment]
@bruno314 please fix quotes and rebase.

**Comment by bruno314** [comment]
All fixed, including my forgotten `"` in setup

**Comment by lnielsen** [comment]
Closing PR as it's been stale for a long time.

**Pull request closed without merging**